redirect/client.js

```diff
@@ -0,0 +1,54 @@
+(function() {
+
+    function apply(context, template) {
+
+        /**
+         * Prevent redirect from occurring while in either the Template Editor or Campaign Editor.
+         */
+        if ((window.frameElement || {}).id === "siteEditorFrame") {
+            return;
+        }
+
+        const currentPage = window.location.hostname + window.location.pathname.replace(/\/$/, "");
+        const targetPage = context.targetPageUrl.replace(/http(s)?\:\/\//, "");
+        const redirectPage = context.redirectUrl.replace(/http(s)?\:\/\//, "");
+        if ((context.targetPageUrl && context.redirectUrl) && (currentPage !== targetPage && currentPage === redirectPage)) {
```

Comment:
Looks good! Line's getting a bit long. Let's split this into two lines.
```suggestion
        if ((context.targetPageUrl && context.redirectUrl) && 
            (currentPage !== targetPage && currentPage === redirectPage)) {
```